jbpm-runtime-manager/src/test/java/org/jbpm/runtime/manager/impl/error/ExecutionErrorHandlingRuntimeManagerTest.java

```diff
@@ -0,0 +1,372 @@
+    }
+    
+    private String strategy;
+       
+    public ExecutionErrorHandlingRuntimeManagerTest(String strategy) {
+        this.strategy = strategy;
+    }
+    
+    private PoolingDataSource pds;
+    private UserGroupCallback userGroupCallback;
+    private EntityManagerFactory emf;
+    private RuntimeManager manager;
+    
+    @Rule 
+    public TestName testName = new TestName();
+    
+    @Before
+    public void setup() {
+        TestUtil.cleanupSingletonSessionId();
+        pds = TestUtil.setupPoolingDataSource();
+        emf = EntityManagerFactoryManager.get().getOrCreate("org.jbpm.persistence.jpa");
+        Properties properties= new Properties();
+        properties.setProperty("mary", "HR");
+        properties.setProperty("john", "HR");
+        userGroupCallback = new JBossUserGroupCallbackImpl(properties);
+        
+        createRuntimeManager();
+    }
+    
+    @After
+    public void teardown() {
+        if (manager != null) {
+            manager.close();
+        }
+        EntityManagerFactoryManager.get().clear();
+        pds.close();
+    }
+    
+    private void createRuntimeManager() {
+        RuntimeEnvironment environment = createEnvironment();        
+        if ("singleton".equals(strategy)) {
+            manager = RuntimeManagerFactory.Factory.get().newSingletonRuntimeManager(environment, "first");
+        } else if ("processinstance".equals(strategy)) {
+            manager = RuntimeManagerFactory.Factory.get().newPerProcessInstanceRuntimeManager(environment, "first");
+        } else if ("request".equals(strategy)) {
+            manager = RuntimeManagerFactory.Factory.get().newPerRequestRuntimeManager(environment, "first");
+        } else if ("case".equals(strategy)) {
+            manager = RuntimeManagerFactory.Factory.get().newPerCaseRuntimeManager(environment, "first");
+        }
+        assertNotNull(manager);        
+    }
+    
+    @Test
+    public void testBasicScriptFailure() {
+            
+        RuntimeEngine runtime1 = manager.getRuntimeEngine(ProcessInstanceIdContext.get());
+        KieSession ksession1 = runtime1.getKieSession();
+        assertNotNull(ksession1);                 
+        
+        try {
+            ksession1.startProcess("BrokenScriptTask");
+            fail("Start process should fail due to broken script");
+        } catch (Throwable e) {
+            // expected
+        }
+        manager.disposeRuntimeEngine(runtime1);
+       
+        ExecutionErrorManager errorManager = ((AbstractRuntimeManager) manager).getExecutionErrorManager();
+        ExecutionErrorStorage storage = errorManager.getStorage();
+        
+        List<ExecutionError> errors = storage.list(0, 10);
+        assertNotNull(errors);
+        assertEquals(1, errors.size());
+        assertExecutionError(errors.get(0), "Process", "BrokenScriptTask", "Hello");
+        
+        // close manager which will close session maintained by the manager
+        manager.close();
```

Comment:
Hi, Maciej, why there is a manual disposing and closing of the engine and the manager when it is already done in the teardown() method? Thanks.